cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -141,21 +143,27 @@ runQueryProtocolParameters (AnyConsensusModeParams cModeParams) network mOutFile
                            readEnvSocketPath
   let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
 
-  anyE@(AnyCardanoEra era) <- determineEra cModeParams localNodeConnInfo
-  let cMode = consensusModeOnly cModeParams
-  sbe <- getSbe $ cardanoEraStyle era
-
-  case toEraInMode era cMode of
-    Just eInMode -> do
-      let query = QueryInEra eInMode
-                    $ QueryInShelleyBasedEra sbe QueryProtocolParameters
-      result <- executeQuery
-                  era
-                  cModeParams
-                  localNodeConnInfo
-                  query
-      writeProtocolParameters mOutFile result
-    Nothing -> left $ ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE
+  result <- liftIO $ executeQueryLocalState localNodeConnInfo Nothing $ \_ntcVersion -> do
+    anyE@(AnyCardanoEra era) <- determineEra2 cModeParams
+
+    case cardanoEraStyle era of
+      LegacyByronEra -> return $ Left ShelleyQueryCmdByronEra
+      ShelleyBasedEra sbe -> do
+        let cMode = consensusModeOnly cModeParams
+        case toEraInMode era cMode of
+          Just eInMode -> do
+            ppResult :: Either EraMismatch ProtocolParameters <- sendMsgQuery $
+              QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryProtocolParameters
+            case ppResult of
+              Right pp -> return (Right pp)
+              Left e -> return (Left (ShelleyQueryCmdEraMismatch e))
```

Comment:
Lots of brackets (minor)

Reply:
I think this line doesn't exist anymore.